adafruit_httpserver/route.py

```diff
@@ -22,6 +22,9 @@ def __hash__(self) -> int:
         return hash(self.method) ^ hash(self.path)
 
     def __eq__(self, other: "_HTTPRoute") -> bool:
+        if self.path[-1] == '*':
+            prepath = self.path[0:len(self.path)-1]
+            return self.method == other.method and other.path.startswith(prepath)
```

Comment:
Simpler. You could also use `self.path.rstrip("*") instead of `self.path[0:-1]`
```suggestion
        if self.path.endswidth("*"):
            return self.method == other.method and other.path.startswith(self.path[0:-1])
```

Reply:
Overwriting eq method this way, it can lead to situations where A == B but B != A.

I have been thinking for a while about wildcards in URLs, already working on a more functional approach to this, that will also allow parsing those values.

Reply:
> Overwriting eq method this way, it can lead to situations where A == B but B != A.

i agree with @michalpokusa's caution about this -- I wasn't sure whether it was something standard in server routing.